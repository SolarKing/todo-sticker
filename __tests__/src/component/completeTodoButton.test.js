// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import CompleteTodoButton from '../../../src/component/complteTodoButton/completeTodoButton';
import { Modal } from 'antd';

jest.mock('antd', () => {
    let Modal = {
        confirm: jest.fn(),
    };
    return {
        Modal
    };
});

describe('CompleteTodoButton', () => {
    let wrapper: ShallowWrapper;
    let handleCompleteTodo = jest.fn();

    beforeAll(() => {
        wrapper = shallow(<CompleteTodoButton handleCompleteTodo={handleCompleteTodo} todoId={10} />);
    });

    it('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should excute modal confirm when clicked', () => {
        wrapper.find('a').simulate('click');
        expect(Modal.confirm.mock.calls.length).toEqual(1);
    });

    it('should execute handleCompleteTodo when confirming successfully', () => {
        wrapper.find('a').simulate('click');
        Modal.confirm.mock.calls[0][0].onOk();
        expect(handleCompleteTodo.mock.calls.length).toEqual(1);
        expect(handleCompleteTodo.mock.calls[0][0]).toEqual(10);
    });

});