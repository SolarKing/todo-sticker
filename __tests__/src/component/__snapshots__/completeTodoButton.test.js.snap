// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompleteTodoButton should render correct 1`] = `
<a
  onClick={[Function]}
>
  Complete
</a>
`;
